test/plugins.js

```diff
@@ -1,42 +1,106 @@
-import * as chai from '../index.js';
+import {use, expect, Should} from '../index.js';
+
+function plugin(chai) {
+  if (chai.Assertion.prototype.testing) return;
+
+  chai.assert.testing = 'successful';
+
+  Object.defineProperty(chai.Assertion.prototype, 'testing', {
+    get: function () {
+      return 'successful';
+    },
+  });
+}
+
+function anotherPlugin(chai) {
+  if (chai.Assertion.prototype.moreTesting) return;
+
+  chai.assert.moreTesting = 'more success';
+
+  Object.defineProperty(chai.Assertion.prototype, 'moreTesting', {
+    get: function () {
+      return 'more success';
+    },
+  });
+}
+
+function brokenPlugin(chai) {
+  chai.overwriteProperty('equal', function (_super) {
+    if (something) {
+      return _super.call(this);
+    }
+    return someOtherThing();
+  });
+}
 
 describe('plugins', function () {
+  it("doesn't crash when there's a bad plugin", function () {
+    expect(() => {
+      use(brokenPlugin).use(brokenPlugin).use(brokenPlugin);
+    }).to.not.throw;
+  });
+
+  describe('should', () => {
+    before(() => {
+      Should();
+    });
+
+    it('basic usage', function () {
+      use(plugin);
```

Comment:
do these tests actually pollute each other? in that `use` in this test will mean the next one already has `plugin` setup?

not sure how you'd solve that though since we can't reset the prototype 🤔  we'd have to use a unique plugin for each test which is also a bit of a pain

Reply:
Aha! Yes they do!

What about I set `configurable: true` in the plugins and then `delete` the properties in the `after`? I think that should work.

Reply:
That could work. Maybe give it a try, hopefully the tests still pass 🙏

Reply:
I ended up leaning into the global nature of plugins and just apply them once in the beginning of tests.